### ISSUE TITLE
Functions related to a deleted game object still work

### ISSUE THREAD

**Issue by einarnordengren** [defold/defold#7666]
**Describe the bug (REQUIRED)**
After calling go.delete, functions related to that game object still work during the same frame delete is called. go.get_id() returns the id, go.exists() returns true and go.get_position() returns the position. Mentioned here: [https://forum.defold.com/t/what-happens-immediately-after-deleting-a-game-object](https://forum.defold.com/t/what-happens-immediately-after-deleting-a-game-object)

**To Reproduce (REQUIRED)**
1. Call go.delete
2. Call go.get_id, go.get_position, go.exists, or any other function related to that game object

**Expected behavior (REQUIRED)**
I would expect the functions related to the game object to stop working or throwing errors about a missing game object immediately after deleting it.

**Defold version (REQUIRED):**
  - Version 1.4.5

**Platforms (REQUIRED):**
 - Platforms: Windows, editor
 - OS: Windows 11

**Comment by JCash** [comment]
It is in fact a delayed delete, and the game object is deleted at the end of the frame.
As such, the rest of the scripts which may interact with the object can still operate on it.

The execution order of the scripts is unordered, so this allows them all to work for that frame.

If we were to change that behavior, I assume a lot of existing games out there would start failing in various ways.
I don't think we'd like that though.

Currently, I'm leaning towards a "won't fix" on this issue.

**Comment by britzl** [comment]
Yeah, this is behaviour that we can't change. And as you say, the object will not be deleted until the end of the frame.
